Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/transform/policy/AdhocQueryTransformHelper.java

```diff
@@ -0,0 +1,202 @@
+            .getAttributeValue().get(0).getContent().get(0));
+        SubjectType subjectType = actualCheckPolicy.getRequest().getSubject().get(0);
+        AttributeType attributeType = subjectType.getAttribute().get(1);
+        assertEquals("http://www.hhs.gov/healthit/nhin#HomeCommunityId", attributeType.getAttributeId());
+        assertEquals("http://www.w3.org/2001/XMLSchema#anyURI", attributeType.getDataType());
+        assertEquals("urn:oid:1.1.1.1.1.1.1", attributeType.getAttributeValue().get(0).getContent().get(0));
+    }
+    
+ 
+    @Test
+    public void testTransformAdhocQueryToCheckPolicy_InValidDirection() {
+        AdhocQueryRequestEventType eventA = new AdhocQueryRequestEventType();
+        eventA.setDirection("in_outbound");
+        CheckPolicyRequestType result = instance.transformAdhocQueryToCheckPolicy(eventA);
+        assertNull("We should expect null checkPolicyEngine ", result);
+        eventA.setDirection(null);
+        assertNull(instance.transformAdhocQueryToCheckPolicy(eventA));
+        assertNull(instance.transformAdhocQueryToCheckPolicy(null));
+    }
+    
+ 
+    @Test
+    public void testTransformAdhocQueryToCheckPolicy_outbound() {
+    	AdhocQueryTransformHelper instance = new AdhocQueryTransformHelper();
+        AdhocQueryRequestEventType eventA = new AdhocQueryRequestEventType();
+     //   eventA.getDirection()= "inbound";
+       
+        eventA.setDirection("outbound");
+        CheckPolicyRequestType result = instance.transformAdhocQueryToCheckPolicy(eventA);
+        assertNotNull(result);
+    }
+
+    @Test
+    public void testTransformAdhocQueryToCheckPolicy_AssertNotNull() {
+    	AdhocQueryTransformHelper instance = new AdhocQueryTransformHelper();
+        AdhocQueryRequestEventType eventA = new AdhocQueryRequestEventType();
+     //   eventA.getDirection()= "inbound";
+       
+        eventA.setDirection("inbound");
+        CheckPolicyRequestType result = instance.transformAdhocQueryToCheckPolicy(eventA);
+        assertNotNull(result);
+    }
+
+    @Test
+    public void testTransformAdhocQueryResponseToCheckPolicy_OutBound() {
+    	AdhocQueryTransformHelper instance = new AdhocQueryTransformHelper();
+    	AdhocQueryResultEventType eventB = new AdhocQueryResultEventType();
+     //   eventA.getDirection()= "inbound";
+        eventB.setDirection("outbound");
+        CheckPolicyRequestType result = instance.transformAdhocQueryResponseToCheckPolicy(eventB);
+        assertNotNull(result);
+    }
+    
+    @Test
+    public void testTransformAdhocQueryResponseToCheckPolicy_InBound() {
+    	AdhocQueryTransformHelper instance = new AdhocQueryTransformHelper();
+    	AdhocQueryResultEventType eventB = new AdhocQueryResultEventType();
+     //   eventA.getDirection()= "inbound";
+        eventB.setDirection("inbound");
+        CheckPolicyRequestType result = instance.transformAdhocQueryResponseToCheckPolicy(eventB);
+        assertNotNull(result);
+    }
+
+    
+}
```

Comment:
Please see this wiki page for code formatter. 
 https://connectopensource.atlassian.net/wiki/display/CONNECTWIKI/Coding+templates+and+formatters